src/pages/ProfilesExplorerView/components/SceneExploreDiffFlameGraph/components/SceneComparePanel/SceneComparePanel.tsx

```diff
@@ -199,6 +207,19 @@ export class SceneComparePanel extends SceneObjectBase<SceneComparePanelState> {
             : TimeRangeWithAnnotationsMode.DEFAULT,
       });
     });
+
+    const timeRangeSub = this.state.$timeRange.subscribeToState((newState, prevState) => {
+      if (newState.from !== prevState.from || newState.to !== prevState.to) {
+        this.updateTitle('');
+      }
+    });
+
+    return {
```

Comment:
**suggestion (non-blocking)**: Automatically clean up subscriptions

I think in scenes when you use `this._subs.add(...)` it will automatically clean up all subscriptions when a scene is deactivated. Anything added with `this.subscribeToEvent(...)` should also be removed automatically:

- [subscribeToEvent()](https://github.com/grafana/scenes/blob/v4.22.0/packages/scenes/src/core/SceneObjectBase.tsx#L114) - [clean up code](https://github.com/grafana/scenes/blob/v4.22.0/packages/scenes/src/core/SceneObjectBase.tsx#L311)
- _subs.add [clean up code](https://github.com/grafana/scenes/blob/v4.22.0/packages/scenes/src/core/SceneObjectBase.tsx#L311)

Reply:
Thank you for this! I always wanted to refactor it ; I'll do it in a future PR.